examples/AnimalAndFlower/model.py

```diff
@@ -0,0 +1,180 @@
+    image[sunflower] = ModuleLearner('emb', module=nn.Linear(output_size, 2), device=device)
+    program = IMLProgram(graph, poi=(image,), inferTypes=['ILP', 'local/argmax'],
+                         loss=MacroAverageTracker(BCEWithLogitsIMLoss(lmbd=0.5)),
+                         metric={'ILP': PRF1Tracker(DatanodeCMMetric()),
+                                 'softmax': PRF1Tracker(DatanodeCMMetric('local/argmax'))})
+    return program
+
+
+
+def train_with_single_network(train_set, validation_set, labelKeys):
+    baseNet = ImageNetwork()
+    classCounts = len(labelKeys)
+    fcModels = {key: nn.Linear(output_size, 2) for key in labelKeys}
+
+    criterion = nn.CrossEntropyLoss()
+    params = list(baseNet.parameters())
+    for key in labelKeys:
+        params += list(fcModels.get(key).parameters())
+
+    optimizer = optim.Adam(params)
+
+    for epoch in range(1000):  # loop over the dataset multiple times
+
+        for i, data in enumerate(tqdm(train_set), 0):
+            optimizer.zero_grad()
+            inputs = data['pixels']
+            del data['pixels']
+            losses = []
+            baseNetResult = baseNet(inputs)
+            for key in labelKeys:
+                output = fcModels[key](baseNetResult)
+                losses.append(criterion(output, torch.tensor([data[key]])))
+
+            torch.autograd.backward(losses)
+            optimizer.step()
+        finalLabels = []
+        finalRes = []
+
+        for i, data in enumerate(tqdm(validation_set), 0):
+            optimizer.zero_grad()
+            inputs = data['pixels']
+            del data['pixels']
+            baseNetResult = baseNet(inputs)
+            labels = []
+            result = []
+            for key in labelKeys:
+                output = fcModels[key](baseNetResult)
+                labels.append(data[key])
+                result.append(int(torch.argmax(output)))
+
+            finalRes.append(np.array(result))
+            finalLabels.append(np.array(labels))
+
+        finalForEach = [{"tp": 0, "tn": 0, "fp": 0, "fn": 0} for i in range(classCounts)]
+        total = {"tp": 0, "tn": 0, "fp": 0, "fn": 0}
+        for results, labels in zip(finalRes, finalLabels):
+            for index, (result, label) in enumerate(zip(results, labels)):
+                if result == label and label == 1:
+                    finalForEach[index]["tp"] += 1
+                    total["tp"] += 1
+                elif result == label and label == 0:
+                    finalForEach[index]["tn"] += 1
+                    total["tn"] += 1
+                elif result != label and label == 1:
+                    finalForEach[index]["fp"] += 1
+                    total["fp"] += 1
+                elif result != label and label == 0:
+                    finalForEach[index]["fn"] += 1
+                    total["fn"] += 1
+
+        def getMetric(data):
+            e = 0
```

Comment:
the framework should provide the evaluation, is this needed?

Reply:
it is not needed anymore